lib/query_components/core_query.rb

```diff
@@ -47,7 +56,10 @@ def payload
       }
     end
 
-    private
+    def payload_for_quoted_phrase
+      groups = [ field_boosts_phrase ]
+      groups.map { |queries| dis_max_query(queries) }
+    end
```

Comment:
This is like this to match the implementation in text_query, which in turn uses a shared function to apply the dis_max.  It might be better written this way, but I'm on the fence, so not going to block merge on this.